controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -223,10 +223,15 @@ func (r *StorageClassRequestReconciler) reconcilePhases() (reconcile.Result, err
 		profileName = r.storageCluster.Spec.DefaultStorageProfile
 	}
 
-	for i := range r.storageCluster.Spec.StorageProfiles {
-		profile := &r.storageCluster.Spec.StorageProfiles[i]
+	storageProfileList := &v1.StorageProfileList{}
+	if err := r.list(storageProfileList, client.InNamespace(r.OperatorNamespace)); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	for _, storageProfile := range storageProfileList.Items {
+		profile := storageProfile
 		if profile.Name == profileName {
-			r.storageProfile = profile
+			r.storageProfile = &profile
```

Comment:
It is preferable to not store information related to a specific request on the reconciler as it is persistent between reconcile.

I will suggest to load all storage profiles into a map key by the profile name in the first reconcile and index into it using the name on the request to retrieve the profile for each request